### PR TITLE
Add TypeScript definition

### PR DESCRIPTION
**Pull request by BendingBender**

### CODE REVIEW

File: index.d.ts [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+		readonly name?: string;
+	}
+>;
+
+/**
+Get a temporary file path you can write to.
+
+@example
+```
+import * as tempy from tempy';
```
Comment (sindresorhus) [inline index.d.ts]: I don't really want to recommend using named imports here. I don't intend to expose them as such when eventually moving to ESM. See: https://github.com/sindresorhus/meta/issues/6

Comment (BendingBender) [reply index.d.ts]: Not sure what you propose instead. Can you give me a hint?

Comment (sindresorhus) [reply index.d.ts]: `import tempy = require('tempy');` for now and `import tempy from 'tempy';` later on when using ESM.

Comment (BendingBender) [reply index.d.ts]: I can change this but you have to know that nothing will forbid users to use the `import * as tempy` syntax even without `esModulesInterop`.

Comment (sindresorhus) [reply index.d.ts]: I’m aware, but at least I’ve shown the recommended way.

**Pull request merged**